### PR TITLE
Implement getBoidsGroupById

### LINKED ISSUE #24: Implement TestData.getBoidsGroupById
Split from #22

### PR DESCRIPTION
**Pull request by rexim**
Close #24

### CODE REVIEW

File: src/Boids.hs [hunk 1]
```diff
@@ -22,7 +22,7 @@ data World = World { worldBoids :: [Boid]
 data Boid = Boid { boidPosition :: Point
                  , boidHeading :: Float
                  , boidSteer :: Float
-                 } deriving Show
+                 } deriving (Show, Eq, Ord)
```
Comment (Minoru) [inline src/Boids.hs:25]: `Eq` for `Boid`? Where's [the comment I asked for two months ago?](https://github.com/tsoding/boids/pull/21#discussion_r91361969) :P

File: src/TestData.hs [hunk 2]
```diff
@@ -52,16 +60,27 @@ circleToBoid element
...
+
 hasId :: Element -> String -> Bool
 hasId element id =
     fromMaybe False $ lookupAttr idAttrQName (elAttribs element) >>= \a -> return (a == id)
 
 isCircleWithId :: String -> Element -> Bool
 isCircleWithId id element = isCircle element && hasId element id
 
+isGroupWithId :: String -> Element -> Bool
+isGroupWithId id element = isGroup element && hasId element id
```
Comment (Minoru) [inline src/TestData.hs]: Consider ```element `hasId` id```—IMHO it's a bit easier to understand.

File: test/TestDataSpec.hs [hunk 3]
```diff
@@ -14,6 +15,23 @@ import Data.Maybe
...
+
+assertBoidsEqual :: Boid -> Boid -> Assertion
+assertBoidsEqual expectedBoid actualBoid = sequence_ assertions
+    where
+      properties = [ (fst . boidPosition, "X coordinates were not equal")
+                   , (snd . boidPosition, "Y coordinates were not equal")
+                   , (boidHeading, "Headings were not equal")
+                   , (boidSteer, "Steers were not equal") ]
+      assertions = map assertBoidsEqual properties
+      assertBoidsEqual (f, message) = assertApproxEqual message errorMargin (f expectedBoid) (f actualBoid)
```
Comment (Minoru) [inline test/TestDataSpec.hs]: Please come up with another name for this function (the one in the `where` clause). It clashes with top-level `assertBoidsEqual`.

**Comment by Minoru** [comment]
I'm done. I believe the first and third comments are critical enough to prevent this from being merged until they're addressed.

(Sorry for individual comments, by the way. I… don't know what happened there.)

File: src/Boids.hs (hunk 1 above)
Comment (rexim) [reply src/Boids.hs:25]: It's impossible to derive it from Ord without Eq

Comment (rexim) [reply src/Boids.hs:25]: I'm not considering this comment blocking. Boids will derive from Eq and Ord until I resolve #27.

Comment (Minoru) [reply src/Boids.hs:25]: > It's impossible to derive it from Ord without Eq

Derive what?

Re #27: got it. Okay then. I'll be watching you ._.

Comment (rexim) [reply src/Boids.hs:25]: > Derive what?

Boid

Comment (Minoru) [reply src/Boids.hs:25]: Ah, you meant that you can't derive `Ord` without `Eq`. Got it.

**Comment by rexim** [comment]
@Minoru thanks for the review!

**Pull request merged**